params/params.go

```diff
@@ -0,0 +1,101 @@
+package params
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/go-chi/chi/v5"
+
+	"github.com/bir/iken/validation"
+)
+
+const errFormat = "%s parameter invalid %s : %q : %s"
+
+func GetString(r *http.Request, name string, required bool) (string, error) {
+	param := chi.URLParam(r, name)
+	if len(param) == 0 {
+		param = r.URL.Query().Get(name)
+	}
+
+	if required && len(param) == 0 {
+		return "", validation.New(name, name+" parameter not found") //nolint: wrapcheck
+	}
+
+	return param, nil
+}
+
+func GetInt32(r *http.Request, name string, required bool) (*int32, error) {
+	s, err := GetString(r, name, required)
+	if err != nil || len(s) == 0 {
+		return nil, err
+	}
+
+	i, err := strconv.ParseInt(s, 10, 32)
+	if err != nil {
+		return nil, validation.New(name, fmt.Sprintf(errFormat, name, "int32", s, err)) //nolint: wrapcheck
```

Comment:
future improvement: it would be nice to refactor validation messages to be error based instead of string based. Then we could wrap this err instead of turning it into a string.

Reply:
Not sure I can think of a use case where having more granularity of the errors delivers any value.

validation errors are automatically treated as Bad Requests by the error handler:

https://github.com/bir/iken/blob/2689e9d50092eea239b376a31878c93e8df2dfe1/httputil/errors.go#L45

My expectation is any error with params would yield an HTTP4XX.  Returning generic errors here would then require the caller to handle that mapping.